agent/tcs/handler/handler_test.go

```diff
@@ -73,6 +73,19 @@ func (*mockStatsEngine) GetTaskHealthMetrics() (*ecstcs.HealthMetadata, []*ecstc
 	return nil, nil, nil
 }
 
+// TestDisableMetrics tests the StartMetricsSession will return immediately if
```

Comment:
just wondering what is the result of breaking the logic we test here? it seems that in that case we will just not return immediately, but i'm not sure whether the test will fail?

Reply:
hrm yea this is a weird one, this looks like it needs a `StartMetricsSession` refactor to be able to have more sensible testing. we can track this as tech debt.

Reply:
nit: add a TODO for this